### PR TITLE
#3768 - Fixing use of  item on the house door

### PR DESCRIPTION
**Pull request by ArturKnopik**
#3768 - Fixing use of  item on the house door

### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### CODE REVIEW

**Comment by marmichalski** [comment]
This is not only house door but also "shop-like" houses where walls are counters. Anyway, that's nitpicking, but shouldn't that feature be somehow dependant of `ONLY_INVITED_CAN_MOVE_HOUSE_ITEMS` config option? And maybe that option should be rename to something more universal? Or maybe this feature should have it's own flag. Having it enable and non-configurable is not a way to go, IMHO.

File: src/actions.cpp [hunk 1]
```diff
@@ -427,6 +427,14 @@ bool Actions::useItem(Player* player, const Position& pos, uint8_t index, Item*
 		showUseHotkeyMessage(player, item, player->getItemTypeCount(item->getID(), subType != item->getItemCount() ? subType : -1));
 	}
 
+	if (HouseTile* houseTile = dynamic_cast<HouseTile*>(item->getTile())) {
+		House* house = houseTile->getHouse();
+		if (house && !house->isInvited(player)) {
+			player->sendCancelMessage(RETURNVALUE_NOTPOSSIBLE);
```
Comment (marmichalski) [inline src/actions.cpp:433]:
```suggestion
			player->sendCancelMessage(RETURNVALUE_PLAYERISNOTINVITED);
```

❓

File: src/actions.cpp [hunk 2]
```diff
@@ -427,6 +427,14 @@ bool Actions::useItem(Player* player, const Position& pos, uint8_t index, Item*
...
 	}
 
+	if (HouseTile* houseTile = dynamic_cast<HouseTile*>(item->getTile())) {
+		House* house = houseTile->getHouse();
+		if (house && !house->isInvited(player)) {
+			player->sendCancelMessage(RETURNVALUE_PLAYERISNOTINVITED);
+			return false;
+		}
+	}
+
```
Comment (nekiro) [inline src/actions.cpp:437]:
```suggestion
if (g_config.getBoolean(ConfigManager::ONLY_INVITED_CAN_MOVE_HOUSE_ITEMS)) {
	if (HouseTile* houseTile = dynamic_cast<HouseTile*>(item->getTile())) {
		House* house = houseTile->getHouse();
		if (house && !house->isInvited(player)) {
			player->sendCancelMessage(RETURNVALUE_PLAYERISNOTINVITED);
			return false;
		}
	}
}
```

**Review by nekiro — CHANGES_REQUESTED** [review]

**Comment by EPuncker** [comment]
tabbing seems to be off by 1 from the rest of the code, other than that, looks good

**Comment by nekiro** [comment]
> tabbing seems to be off by 1 from the rest of the code, other than that, looks good

Might be, my tabbing skills in www github are bad xD

**Comment by ArturKnopik** [comment]
cleaned

File: src/actions.cpp [hunk 3]
```diff
@@ -427,6 +427,15 @@ bool Actions::useItem(Player* player, const Position& pos, uint8_t index, Item*
...
 
+	if (g_config.getBoolean(ConfigManager::ONLY_INVITED_CAN_MOVE_HOUSE_ITEMS)) {
+		if (HouseTile* houseTile = dynamic_cast<HouseTile*>(item->getTile())) {
+			House* house = houseTile->getHouse();
+			if (house && !house->isInvited(player)) {
+				player->sendCancelMessage(RETURNVALUE_PLAYERISNOTINVITED);
+				return false;
+			}
+		}
+	}
```
Comment (nekiro) [inline src/actions.cpp:438]: new line here please

**Review by nekiro — APPROVED** [review]

**Review by EPuncker — APPROVED** [review]

**Pull request merged**